blueflood-http/src/main/java/com/rackspacecloud/blueflood/tracker/Tracker.java

```diff
@@ -38,14 +38,42 @@
 
     private static final Logger log = LoggerFactory.getLogger(Tracker.class);
     private static final String trackerName = String.format("com.rackspacecloud.blueflood.tracker:type=%s", Tracker.class.getSimpleName());
-    private static final Pattern patternGetTid = Pattern.compile( "/v\\d+\\.\\d+/([^/]+)/.*" );
+    private final Pattern patternGetTid = Pattern.compile( "/v\\d+\\.\\d+/([^/]+)/.*" );
 
-    static Set tenantIds = new HashSet();
-    static boolean isTrackingDelayedMetrics = false;
-    static Set<String> metricNames = new HashSet<String>();
+    // Tracker is a singleton
+    private static Tracker instance;
+    private static boolean isRegistered = false;
 
-    public Tracker() {
-        registerMBean();
+    private Set tenantIds = new HashSet();
+    private boolean isTrackingDelayedMetrics = false;
+    private Set<String> metricNames = new HashSet<String>();
+
+    // private constructor for singleton
+    private Tracker(){}
+
+    public static synchronized void register() {
```

Comment:
@VinnyQ , why don't we just make people call it like this?
```
Tracker.getInstance().register()
```

It is more explicit. When I read a code like that, the code tells me two information:
* Tracker class is a singleton
* register() is an instance method that registers Tracker to some registry.

Compare that to call like this:
```
Tracker.register()
```
The instance creation is hidden and implicit. It's less clear of what this Tracker class is doing.